rusk-wallet/src/wallet/transaction.rs

```diff
@@ -55,10 +55,10 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
         let refund_pk = self.shielded_key(sender_idx)?;
 
         let inputs = state
-            .inputs(sender_idx, amt + gas.limit * gas.price)
+            .tx_input_notes(sender_idx, amt + gas.limit * gas.price)
```

Comment:
Ideally I would like the arithmetic is done in the function and not in the function call argument
```
.tx_input_notes(sender_idex, amt, gas.limit, gas.price);
```

This way its less confusing when calling the function

Reply:
I disagree. The function pick the notes for a specified cost-value. Calculating the cost that the notes need to cover is the responsible of the caller of the function.
If you want, to clear things up here, I can introduce a variable name like this:
```rust
let tx_costs = amt + gas.limit * gas.price; 
let inputs = state
    .tx_input_notes(sender_idx, tx_cost)
```

Reply:
I see what you mean but would still like to keep it as is. 
Because if we split the cost up into multiple values we would also need to add the deposit as a parameter.

This would lead to a signature like this:
```rust
    pub(crate) async fn tx_input_notes(
        &self,
        index: u8,
        transfer_value: u64,
        deposit: u64,
        gas: &Gas,
    ) -> Result<Vec<(Note, NoteOpening, BlsScalar)>, Error> {

```

If we call that function we would need to create 0-value variables for transfer and deposit if they are not used in the transaction. This to me adds more confusion than it would solve.

Reply:
@moCello what about doing the calculation on the line above and feeding it to the function as a variable instead of directly as a parameter argument?

It's an extra line of code for each call in this file but from a readability point of view probably better.